src/Analysis/Ast/Impl/Specializations/Typing/Types/NamedTupleType.cs

```diff
@@ -27,20 +27,25 @@ internal sealed class NamedTupleType : TypingTupleType, ITypingNamedTupleType {
         /// <summary>
         /// Creates type info for a strongly-typed tuple, such as Tuple[T1, T2, ...].
         /// </summary>
-        public NamedTupleType(string tupleName, IReadOnlyList<string> itemNames, IReadOnlyList<IPythonType> itemTypes, IPythonInterpreter interpreter)
-            : base(itemTypes, interpreter) {
+        public NamedTupleType(string tupleName, IReadOnlyList<string> itemNames, IReadOnlyList<IPythonType> itemTypes, IPythonModule declaringModule, IPythonInterpreter interpreter)
+            : base(itemTypes, declaringModule, interpreter) {
             TupleName = tupleName ?? throw new ArgumentNullException(nameof(tupleName));
             ItemNames = itemNames;
 
             var typeNames = itemTypes.Select(t => t.IsUnknown() ? string.Empty : t.Name);
             var pairs = itemNames.Zip(typeNames, (name, typeName) => string.IsNullOrEmpty(typeName) ? name : $"{name}: {typeName}");
             Name = CodeFormatter.FormatSequence(tupleName, '(', pairs);
+
+            typeNames = itemTypes.Select(t => t.IsUnknown() ? string.Empty : t.QualifiedName);
+            pairs = itemNames.Zip(typeNames, (name, typeName) => string.IsNullOrEmpty(typeName) ? name : $"{name}: {typeName}");
+            QualifiedName = CodeFormatter.FormatSequence($"{declaringModule}:{tupleName}", '(', pairs);
         }
```

Comment:
Adding special calculation of qualified name to all specialized types from `typing`